fe/fe-core/src/test/java/com/starrocks/sql/plan/JoinTest.java

```diff
@@ -2158,4 +2163,97 @@ public void testColocateJoinWithDiffPredicateOrders() throws Exception {
                 "  |  equal join conjunct: 22: cast = 11: t1a");
         FeConstants.runningUnitTest = false;
     }
+
+    @Test
+    public void tesAutoEstimateDopForLocalBucketAndColocateJoin() throws Exception {
+        final int numCPUs = 8;
+        new MockUp<BackendCoreStat>() {
+            @Mock
+            public int getAvgNumOfHardwareCoresOfBe() {
+                return numCPUs;
+            }
+        };
+
+        // Enable DopAutoEstimate.
+        SessionVariable sessionVariable = connectContext.getSessionVariable();
+        Assert.assertNotNull(sessionVariable);
+        sessionVariable.setEnablePipelineEngine(true);
```

Comment:
The session variable value need to set origin value.

Reply:
Yes, it is set to origin value at the tail of this test function.